mitodl/ocw-hugo-themes
# add fields theme

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes https://github.com/mitodl/ocw-hugo-themes/issues/669


#### What's this PR do?
This PR adds a new theme called `fields`.  It allows you to create a site focused on a field of study, essentially making a home page out of a collection.


#### How should this be manually tested?
 - Create a site in [`ocw-studio-rc`](https://ocw-studio-rc.odl.mit.edu/sites) using the `mit-fields` starter.  This has been manually created in RC for testing purposes and the `ocw-hugo-projects` PR has not been created yet to codify it.
 - Clone `ocw-hugo-projects` on the `cg/mit-fields` branch
 - Visit the "Subfields" section and define as many subfields as you want, adding a list of courses to each one
 - Visit the "Field" section and fill out the info about the field site you are creating, selecting a featured Subfield and adding as many other subfields as you want
 - When you are done editing, publish your site.  Publishing will fail because this PR has not yet merged and the `ocw-hugo-projects` PR has not been created yet, but we don't need that for testing.
 - Find your site at https://github.mit.edu/ocw-content-rc/ and clone it locally
 - Set the following environment variables in your `.env` file, replacing `/path/to` with your path to the repos indicated and `your-field-slug` with the folder your course was cloned into in a previous step:
   - `FIELDS_HUGO_CONFIG_PATH=/path/to/ocw-hugo-projects/mit-fields/config.yaml`
   - `FIELDS_CONTENT_PATH=/path/to/ocw-content-rc/your-field-slug`
   - `STATIC_API_BASE_URL=https://ocw-draft-qa.global.ssl.fastly.net/` (for loading content from a static API for the courses linked in your Subfields)
 - Start the site with `npm run start:fields` and visit it at http://localhost:3000
 - Ensure that your field site displays properly and that you can visit the courses linked
 - Read the readme changes and ensure they make sense


#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/12089658/166800522-9faffda9-c99f-4760-8b7c-a90d1ef9ade9.png)
![image](https://user-images.githubusercontent.com/12089658/166800667-873cd324-a3da-4cf6-ac37-111394ee2c7e.png)
